Project/Source/Components/UI/ComponentTransform2D.cpp

```diff
@@ -25,40 +25,31 @@
 #define JSON_TAG_ANCHOR_X "AnchorX"
 #define JSON_TAG_ANCHOR_Y "AnchorY"
 
-std::array<AnchorPreset, 16> ComponentTransform2D::anchorPresets = {
-	AnchorPreset {AnchorPreset::AnchorPresetType::TOP_VERTICAL_LEFT_HORIZONTAL, float2(0.0f, 1.0f), float2(0.0f, 1.0f)},
-	AnchorPreset {AnchorPreset::AnchorPresetType::TOP_VERTICAL_CENTER_HORIZONTAL, float2(0.5f, 1.0f), float2(0.5f, 1.0f)},
-	AnchorPreset {AnchorPreset::AnchorPresetType::TOP_VERTICAL_RIGHT_HORIZONTAL, float2(1.0f, 1.0f), float2(1.0f, 1.0f)},
-	AnchorPreset {AnchorPreset::AnchorPresetType::TOP_VERTICAL_STRETCH_HORIZONTAL, float2(0.0f, 1.0f), float2(1.0f, 1.0f)},
-
-	AnchorPreset {AnchorPreset::AnchorPresetType::MIDDLE_VERTICAL_LEFT_HORIZONTAL, float2(0.0f, 0.5f), float2(0.0f, 0.5f)},
-	AnchorPreset {AnchorPreset::AnchorPresetType::MIDDLE_VERTICAL_CENTER_HORIZONTAL, float2(0.5f, 0.5f), float2(0.5f, 0.5f)},
-	AnchorPreset {AnchorPreset::AnchorPresetType::MIDDLE_VERTICAL_RIGHT_HORIZONTAL, float2(1.0f, 0.5f), float2(1.0f, 0.5f)},
-	AnchorPreset {AnchorPreset::AnchorPresetType::MIDDLE_VERTICAL_STRETCH_HORIZONTAL, float2(0.0f, 0.5f), float2(1.0f, 0.5f)},
-
-	AnchorPreset {AnchorPreset::AnchorPresetType::BOTTOM_VERTICAL_LEFT_HORIZONTAL, float2(0.0f, 0.0f), float2(0.0f, 0.0f)},
-	AnchorPreset {AnchorPreset::AnchorPresetType::BOTTOM_VERTICAL_CENTER_HORIZONTAL, float2(0.5f, 0.0f), float2(0.5f, 0.0f)},
-	AnchorPreset {AnchorPreset::AnchorPresetType::BOTTOM_VERTICAL_RIGHT_HORIZONTAL, float2(1.0f, 0.0f), float2(1.0f, 0.0f)},
-	AnchorPreset {AnchorPreset::AnchorPresetType::BOTTOM_VERTICAL_STRETCH_HORIZONTAL, float2(1.0f, 0.0f), float2(0.0f, 0.0f)},
-
-	AnchorPreset {AnchorPreset::AnchorPresetType::STRETCH_VERTICAL_LEFT_HORIZONTAL, float2(0.0f, 0.0f), float2(0.0f, 1.0f)},
-	AnchorPreset {AnchorPreset::AnchorPresetType::STRETCH_VERTICAL_CENTER_HORIZONTAL, float2(0.5f, 0.0f), float2(0.5f, 1.0f)},
-	AnchorPreset {AnchorPreset::AnchorPresetType::STRETCH_VERTICAL_RIGHT_HORIZONTAL, float2(1.0f, 0.0f), float2(1.0f, 1.0f)},
-	AnchorPreset {AnchorPreset::AnchorPresetType::STRETCH_VERTICAL_STRETCH_HORIZONTAL, float2(0.0f, 0.0f), float2(1.0f, 1.0f)},
-};
-
 void ComponentTransform2D::Update() {
 	CalculateGlobalMatrix();
 }
 
 void ComponentTransform2D::OnEditorUpdate() {
 	float3 editorPos = position;
 
-	ImGui::TextColored(App->editor->titleColor, "Position (X,Y,Z)");
+	ImGui::TextColored(App->editor->titleColor, "Transformation (X,Y,Z)");
 	if (ImGui::DragFloat3("Position", editorPos.ptr(), App->editor->dragSpeed2f, -inf, inf)) {
 		SetPosition(editorPos);
 	}
 
+	float3 scl = scale;
+	if (ImGui::DragFloat3("Scale", scl.ptr(), App->editor->dragSpeed2f, 0, inf)) {
+		SetScale(scl);
+	}
+
+	float3 rot = localEulerAngles;
+	/*if (ImGui::DragFloat3("Rotation", rot.ptr(), App->editor->dragSpeed2f, -inf, inf)) {
+		SetRotation(rot);
+	}*/
```

Comment:
Debug comment. This should be either removed or marked as TODO

Reply:
Ok! Fixed in the last commit ;)